x-pack/plugins/alerting/server/application/rule/methods/bulk_untrack/bulk_untrack_alerts.test.ts

```diff
@@ -0,0 +1,101 @@
+import { alertsServiceMock } from '../../../../alerts_service/alerts_service.mock';
+import { ALERT_RULE_UUID, ALERT_UUID } from '@kbn/rule-data-utils';
+
+const taskManager = taskManagerMock.createStart();
+const ruleTypeRegistry = ruleTypeRegistryMock.create();
+const unsecuredSavedObjectsClient = savedObjectsClientMock.create();
+const encryptedSavedObjects = encryptedSavedObjectsMock.createClient();
+const authorization = alertingAuthorizationMock.create();
+const actionsAuthorization = actionsAuthorizationMock.create();
+const auditLogger = auditLoggerMock.create();
+const logger = loggerMock.create();
+const internalSavedObjectsRepository = savedObjectsRepositoryMock.create();
+
+const alertsService = alertsServiceMock.create();
+
+const kibanaVersion = 'v8.2.0';
+const createAPIKeyMock = jest.fn();
+const rulesClientParams: jest.Mocked<ConstructorOptions> = {
+  taskManager,
+  ruleTypeRegistry,
+  unsecuredSavedObjectsClient,
+  authorization: authorization as unknown as AlertingAuthorization,
+  actionsAuthorization: actionsAuthorization as unknown as ActionsAuthorization,
+  spaceId: 'default',
+  namespace: 'default',
+  getUserName: jest.fn(),
+  createAPIKey: createAPIKeyMock,
+  logger,
+  internalSavedObjectsRepository,
+  encryptedSavedObjectsClient: encryptedSavedObjects,
+  getActionsClient: jest.fn(),
+  getEventLogClient: jest.fn(),
+  kibanaVersion,
+  auditLogger,
+  maxScheduledPerMinute: 10000,
+  minimumScheduleInterval: { value: '1m', enforce: false },
+  isAuthenticationTypeAPIKey: jest.fn(),
+  getAuthenticationAPIKey: jest.fn(),
+  getAlertIndicesAlias: jest.fn(),
+  alertsService,
+};
+
+describe('bulkUntrackAlerts()', () => {
+  let rulesClient: RulesClient;
+  beforeEach(async () => {
+    rulesClient = new RulesClient(rulesClientParams);
+  });
+
+  it('should untrack alert documents and update task states', async () => {
+    alertsService.setAlertsToUntracked.mockResolvedValueOnce([
+      {
+        [ALERT_RULE_UUID]:
+          'did you know that you can put whatever you want into these mocked values',
+        [ALERT_UUID]: "it's true",
+      },
+    ]);
+
+    await rulesClient.bulkUntrackAlerts({
+      indices: [
+        'she had them apple bottom jeans (jeans)',
+        'boots with the fur (with the fur)',
+        'the whole club was lookin at her',
+        'she hit the floor (she hit the floor)',
+        'next thing you know',
+        'shawty got low, low, low, low, low, low, low, low',
+      ],
+      alertUuids: [
+        'you wake up late for school, man, you dont wanna GO',
+        'you ask your mom, please? but she still says NO',
+        'you missed two classes and no homeWORK',
+        'but your teacher preaches class like youre some kinda JERK',
+        'you gotta fight',
+        'for your right',
+        'to paaaaaaaaaarty',
+      ],
+    });
+
+    expect(alertsService.setAlertsToUntracked).toHaveBeenCalledTimes(1);
+    expect(taskManager.bulkUpdateState).toHaveBeenCalledWith(
```

Comment:
Can we update this test to test that we are correctly removing the expected values from the task manager state? It would be even better to extract the logic out into a function so we can test that we're handling different scenarios for task manager state. If we mess up the task manager state at this point, it is very hard to recover.

Reply:
Ok but fair warning, every additional unit test I write, the mock data values will become increasingly unhinged 😀 🙃